### ISSUE TITLE
Azure Service Bus - Topic Subscription Name

### ISSUE THREAD

**Issue by DaleyKD** [dapr/dotnet-sdk#941]
BRAND new to Dapr (just started looking today).

Our current infrastructure has already set up a Service Bus Namespace with a single topic (_also many queues; that's for later_). Then in that topic, we have 8 subscriptions. These are named SIMILAR to what we call the microservice. Then, within each subscription, the `$Default` filter is removed, and the name of the event is added as a correlation filter.

I see where I can specify the `consumerID` (_which appears to be the name of the subscription_) via the component YAML, but how could I do all of this via code?

**Comment by brettveenstra** [comment]
I don't believe a Subscription filtering is available to do what you're asking.  If you let Dapr manage the service bus setup (`disableEntityManagement="true"`) , Dapr-ized applications that subscribe to the topic in your component file will automatically create a subscription.

If that is not acceptable, perhaps you can describe your use case a little further?  Are you looking to only send a certain ContentType to Dapr for example?

**Comment by halspang** [comment]
@DaleyKD - I believe the consumerID defaults to your app ID, but I think that's all we really have. If all you want to do is filter messages, we do have the ability to do that, but I don't think we do on the correlation ID. I'd also love to hear more about your use case.

**Comment by eduards-vavere** [comment]
We have the same architecture. Any movement on this?
